modules/perception/base/file_util.cc

```diff
@@ -0,0 +1,305 @@
+        std::vector<std::string>* files) {
+    if (!Exists(path)) {
+        AINFO << path << " not exist.";
+        return;
+    }
+    boost::filesystem::recursive_directory_iterator itr(path);
+    while (itr != boost::filesystem::recursive_directory_iterator()) {
+        try {
+            if (suffix.empty() ||
+                boost::algorithm::ends_with(itr->path().string(), suffix)) {
+                files->push_back(itr->path().string());
+            }
+            ++itr;
+        } catch (const std::exception & ex) {
+            AWARN << "Caught execption: " << ex.what();
+            continue;
+        }
+    }
+}
+void FileUtil::GetFileList(const string& path, vector<string>* files) {
+    GetFileList(path, "", files);
+}
+
+string FileUtil::GetAbsolutePath(const string& prefix,
+                                 const string& relative_path) {
+  if (relative_path.empty()) {
+    return prefix;
+  }
+  if (prefix.empty()) {
+    return relative_path;
+  }
+  string result = prefix;
+  if (relative_path[0] == '/') {
+    return relative_path;
+  }
+  if (prefix[prefix.length() - 1] != '/') {
+    result.append("/");
+  }
+  return result.append(relative_path);
+}
+
+void FileUtil::GetFileName(const string& file, string* name) {
+  size_t pos_left = file.find_last_of('/');
+  size_t pos_right = file.find_last_of('.');
+  if (pos_right == string::npos) {
+    *name = file.substr(pos_left + 1);
+  } else {
+    *name = file.substr(pos_left + 1, pos_right - pos_left - 1);
+  }
+}
+
+bool FileUtil::CompareFileByDigital(const string& file_left,
+                                    const string& file_right) {
+  return CompareFile(file_left, file_right, FCT_DIGITAL);
+}
+
+bool FileUtil::CompareFileByLexicographical(const string& file_left,
+                                            const string& file_right) {
+  return CompareFile(file_left, file_right, FCT_LEXICOGRAPHICAL);
+}
+
+// private functions
+bool FileUtil::CompareFile(const string& file_left, const string& file_right,
+                           FileCompareType type) {
+  string name_left;
+  GetFileName(file_left, &name_left);
+  string name_right;
+  GetFileName(file_right, &name_right);
+  switch (type) {
+    case FCT_DIGITAL:
+      return atoll(name_left.c_str()) < atoll(name_right.c_str());
+    case FCT_LEXICOGRAPHICAL:
+      return std::lexicographical_compare(name_left.begin(), name_left.end(),
+                                          name_right.begin(), name_right.end());
+    default:
+      AERROR << "Unknown compare type!";
+  }
+  return true;
+}
+
+int FileUtil::NumLines(const std::string& filename) {
+  std::ifstream ifs(filename.c_str());
+  return ifs.good()
+             ? static_cast<int>(count(istreambuf_iterator<char>(ifs),
+                                      istreambuf_iterator<char>(), '\n')) +
+                   1
+             : -1;
+}
+
+}  // namespace base
+}  // namespace perception
+}  // namespace apollo
```

Comment:
Remove redundant functions which are already in https://github.com/ApolloAuto/apollo/blob/master/cyber/common/file.h

Reply:
yea, but i think we should move this out of cyber since it is not cybertron specific. shall we consolidate them together later ?

Reply:
i guess i will do the modification at cyber first and then try moving it out.

Reply:
It was ever in Apollo modules/common. But to follow the rules of:
1. No repeated code.
2. It is OK for Apollo to depend on Cyber, but Cyber should not depend on Apollo. So it's migrated into Cyber.
Sorry for the trouble, but as of now we are most likely to stick to this. If there are new utils, it's OK to add in perception module utils, but redundant functions are not allowed.

Reply:
agreed. will modify based on cyber first